sdk/core/azure-core-http-httpurlconnection/src/main/java/com/azure/core/http/httpurlconnection/SocketClient.java

```diff
@@ -0,0 +1,153 @@
+package com.azure.core.http.httpurlconnection;
+
+import com.azure.core.http.HttpHeader;
+import com.azure.core.http.HttpHeaderName;
+import com.azure.core.http.HttpHeaders;
+import com.azure.core.http.HttpRequest;
+
+import javax.net.ssl.SSLSocket;
+import javax.net.ssl.SSLSocketFactory;
+import java.io.*;
+import java.net.ProtocolException;
+import java.net.Socket;
+import java.net.URL;
+import java.util.*;
+
+// A socket client to be used to handle PATCH requests
+
+public class SocketClient {
```

Comment:
This also does not need to be public API. You can probably make it package-private to reduce the visibility.